keel-web/src/main/kotlin/com/netflix/spinnaker/keel/rest/ResourceController.kt

```diff
@@ -99,8 +100,8 @@ class ResourceController(
     consumes = [APPLICATION_JSON_VALUE, APPLICATION_YAML_VALUE],
     produces = [APPLICATION_JSON_VALUE, APPLICATION_YAML_VALUE]
   )
-  @PreAuthorize("@authorizationSupport.userCanModifySpec(#resource.metadata[serviceAccount], #resource.spec)")
-  fun diff(@RequestBody resource: SubmittedResource<*>): DiffResult {
+  @PreAuthorize("@authorizationSupport.userCanModifySpec(#user, #resource.spec)")
+  fun diff(@RequestHeader("X-SPINNAKER-USER") user: String, @RequestBody resource: SubmittedResource<*>): DiffResult {
```

Comment:
Ad-hoc diff also uses Spinnaker user header rather than the service account from the resource that is submitted (as there isn't one any longer)